contracts/protocore/SelfProtocore.sol

```diff
@@ -70,4 +91,142 @@ contract SelfProtocore is MasterCopyNonUpgradable, GenesisSelfProtocore, Validat
             genesisAuxiliaryTargetBlockNumber
         );
     }
+
+
+    /* External Functions. */
+
+    /**
+     * @notice It proposes a valid link to be voted later by active validators.
+     *         It emits LinkProposed event.
+     *
+     * @dev Function requires :
+     *          - parent vote messagehash must not be 0.
+     *          - target block hash must not be 0.
+     *          - source origin observation must not be 0.
+     *          - source kernel hash must not be 0.
+     *          - source kernel hash must not be 0.
+     *          - source committee lock must not be 0.
+     *          - open kernel hash must be same as source kernel hash.
+     *
+     * @param _parentVoteMessageHash Vote message message of the parent metablock.
+     * @param _targetBlockHash Hash of block at target chain.
+     * @param _targetBlockNumber Block number at target.
+     * @param _sourceOriginObservation Observation of the origin chain.
+     * @param _sourceKernelHash Hash of kernel at origin chain.
+     * @param _sourceDynasty Dynasty of origin chain.
+     * @param _sourceAccumulatedGas Accumulated gas at origin chain.
+     * @param _sourceCommitteeLock Committee lock at source.
```

Comment:
The committee lock that hashes the transaction root on the auxiliary chain.

Reply:
Done.